deploy/example/azdiskschedulerextender/csi-azure-disk-scheduler-extender.yaml

```diff
@@ -0,0 +1,197 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: azdiskschedulerextender
+  namespace: kube-system
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: azdiskschedulerextender-role
+rules:
+  - apiGroups: [""]
+    resources: ["endpoints"]
+    verbs: ["get", "update"]
+  - apiGroups: [""]
+    resources: ["configmaps"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["create", "patch", "update"]
+  - apiGroups: [""]
+    resources: ["endpoints"]
+    verbs: ["create"]
+  - apiGroups: [""]
+    resourceNames: ["kube-scheduler"]
+    resources: ["endpoints"]
+    verbs: ["delete", "get", "patch", "update"]
+  - apiGroups: [""]
+    resources: ["nodes"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: [""]
+    resources: ["pods"]
+    verbs: ["delete", "get", "list", "watch"]
+  - apiGroups: [""]
+    resources: ["bindings", "pods/binding"]
+    verbs: ["create"]
+  - apiGroups: [""]
+    resources: ["pods/status"]
+    verbs: ["patch", "update"]
```

Comment:
>`   - apiGroups: [""]`
>`    resources: ["bindings", "pods/binding"]`
>`    verbs: ["create"]`
>`  - apiGroups: [""]`
>`    resources: ["pods/status"]`
>`    verbs: ["patch", "update"]`

Does the scheduler really need to affect pod state or bindings? I would think a scheduler needs a read-only view. Same with `endpoints` above.

Reply:
Yeah, binding is one of the kube-scheduler functions so that looks correct.

That said, let me provide some more context, after the last iteration, I tried to work with system:kube-scheduler. But this role seems to have an issue. Discussed here:
https://github.com/kubernetes/kubernetes/issues/71752
https://github.com/kubernetes/kubernetes/issues/72473

Most of these fixes should have made it to v1.17 and higher, but when it try to dump the "system:kube-scheduler" ClusterRole in my aks deployment it doesn't have access to configMap and that causes scheduler to fail.

I've crossed check system:kube-scheduler ClusterRole in aks deployment and the role we are creating above and there are some differences, one being ConfigMap and other significant one is that we are giving patch/delete access to the endpoints resources. But even that seems to be present in some version of k8s. (they look to have been making changes, endpoints were needed for leases, but now they are using separate APIs)
I can try reconcile azdiskschedulerextender-role with system:kube-scheduler in the code, and make azdiskschedulerextender-role=system:kube-scheduler+configMaps, but I am afraid if that would lead to some issues down the road, given it doesn't have the cofigmaps:
https://github.com/kubernetes/kubernetes/blob/9d99dbc357ed85bcc963fc4bab8a4a3089c910b2/plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go#L525

So, I am inclined to keep this as is for the time being, the ClusterRole we have appears to be a union of rules which were in the previous versions, latest version in k8s code and the ones missing in the code.

Let me know what you think.

Reply:
Ah, thanks for the information and additional context. The bindings and endpoint access rights make sense to me now. What about patch/update access to the pod/status, though? Is that necessary?

Reply:
As we discussed offline, the K8s scheduler requests this permission and extenders may need to update pod state in response to observed conditions. We can scrub this later if we find ours doesn't need this and it doesn't affect permissions needed by other containers in the pod.